core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2770,18 +2770,13 @@ wasm_runtime_invoke_native_raw(WASMExecEnv *exec_env, void *func_ptr,
             {
                 uint32 externref_idx = *argv_src++;
 
-                if (is_aot_func) {
-                    *(uint32 *)argv_dst = externref_idx;
-                }
-                else {
-                    void *externref_obj;
+                void *externref_obj;
```

Comment:
Seems we should replace `if (is_aot_func)` with `if (!signature)` but not remove it